### PR TITLE
Don't suggest change to users that will break dags on 1.10.x

### LINKED ISSUE #12895: from airflow.operators.python import PythonOperator does not work
This is not necessarily a bug in core Airflow, but the upgrade-check scripts recommend this as a solution when the old 1.10.x version of importing the python operator is used. 

So, there is a mismatch between the core Airflow code and the recommendations given in the upgrade check. 

**Apache Airflow version**:

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`):

**Environment**:

- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release):
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:

**What you expected to happen**:

**How to reproduce it**:

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by ashb**
We don't have airflow.operators.python etc in 1.10.x, so suggesting
users change this would break their dags currently.

Closes #12895

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/upgrade/rules/import_changes.py [hunk 1]
```diff
@@ -39,7 +39,9 @@ class ImportChange(
     )
 ):
     def info(self, file_path=None):
-        msg = "Using `{}` should be replaced by `{}`".format(self.old_path, self.new_path)
+        msg = "Using `{}` should be replaced by `{}`".format(
+            self.old_path, self.new_path
+        )
```
Comment (ashb) [inline airflow/upgrade/rules/import_changes.py:44]: (I black-formatted this)

**Review by turbaszek — APPROVED** [review]

**Review by kaxil — APPROVED** [review]

**Comment by ashb** [comment]
YOLO merging, I tested this locally, and it does work.

File: airflow/upgrade/rules/import_changes.py [hunk 2]
```diff
@@ -80,10 +82,30 @@ class ImportChangesRule(BaseRule):
         "https://github.com/apache/airflow#backport-packages"
     )
 
+    current_airflow_version = Version(__import__("airflow").__version__)
+
+    if current_airflow_version < Version("2.0.0"):
+
+        def _filter_incompatible_renames(arg):
+            new_path = arg[1]
```
Comment (eladkal) [inline airflow/upgrade/rules/import_changes.py:90]: Shouldn't this be `new_path = arg[0]` ?
`arg[0]` is the new path while `arg[1]` is the old path.
We want to filter out new paths that lands in ariflow.operator/sensor/hooks

Comment (kaxil) [reply airflow/upgrade/rules/import_changes.py:90]: cc @Dr-Denzy

Comment (drdenzy) [reply airflow/upgrade/rules/import_changes.py:90]: I will look into it.

Comment (eladkal) [reply airflow/upgrade/rules/import_changes.py:90]: @Dr-Denzy you can use the DAG example in https://github.com/apache/airflow/issues/14258 to verify it

**Pull request merged**